docs/units/ref_api.rst

```diff
@@ -2,6 +2,7 @@ Reference/API
 *************
 
 .. automodapi:: astropy.units.quantity
+   :noindex:
```

Comment:
Would be good to know *why* this now becomes necessary (and are you sure it is, even after the rebase?). In principle, I though sphinx could distinguish well between places where a class is defined and the nominal location from which it should be imported (`u.Quantity` vs `u.quantity.Quantity`). Is the problem that there now is a third place where `Quantity` is defined? If so, is this necessary? And most importantly, have you checked that this doesn't break links elsewhere in the documentation?

Reply:
I'll give it a look. Purely a phenomenological fix.

Reply:
Pushed a commit trying to remove the :noindex:. If it works, great! If not, then I'll take a look at whether the docs are effected by adding the :noindex:

Reply:
It does appear to be necessary

Reply:
I'll check the docs to make sure it looks good.

Reply:
Looks the same!
https://astropy--15852.org.readthedocs.build/en/15852/units/ref_api.html

Reply:
Indeed, and the equivalencies too. So, maybe we are stuck with going with this for now. I also checked that a search still gives both locations in the documentation (I tried finding an actual index, but don't think we produce one!).

Anyway, could you add comments both here and for the equivalencies entry that this `:noindex:` is added to solve sphinx problems related to typing? I think we also should have an issue about that, where one includes the error message that one gets without the `:noindex:` part. And ideally a link to a sphinx issue...

Reply:
Sounds good. I'll get to that soon.

 - [ ] comments in the code explaining the issue
 - [ ] issue in Astropy reminding us to fix this
 - [ ] possibly an issue in Sphinx, though making a MWE will be a lot of work...

Reply:
Sphinx issue already exists: https://github.com/sphinx-doc/sphinx/issues/9813

Reply:
Astropy issue: https://github.com/astropy/astropy/issues/16473

Reply:
Same here - thanks for adding the clear comment and opening the issues!